.github/workflows/pull_request.yml

```diff
@@ -94,3 +94,49 @@ jobs:
 
       - name: Run universe tests
         run: ./mvnw -pl :universe verify -Puniverse -Dmaven.javadoc.skip=true
+
+  compatibility_test:
+    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - uses: satackey/action-docker-layer-caching@v0.0.8
+        continue-on-error: true
+
+      - name: Prepare top of trunk docker image
+        run: ./mvnw clean install -DskipTests -Pdocker -Dmaven.javadoc.skip=true -T 1C
+
+      - name: Pull base docker image for cross version env
+        run: docker pull corfudb/corfu-server:BASE
```

Comment:
we need to double-check that 'BASE' can be used, for now you need to deploy docker container locally and the image name will be `corfudb/corfu-server:${corfu_version}`

Reply:
tag BASE now is built temporarily for the branch, under the `compatibility` profile in pom, which represents the top of trunk Corfu server version. tag `0.3.1-SNAPSHOT` is pulled down from dockerhub, which represents parent version and will be updated automatically after https://github.com/CorfuDB/CorfuDB/pull/2902 is merged. Also, by doing this we keep the `0.3.1-SNAPSHOT` consistently refers to parent version in corfudb-cloud repo in version compatibility use case.

Reply:
as discussed offline, we will use 'latest' tag